lib/AST/RequirementMachine/RequirementLowering.cpp

```diff
@@ -802,15 +803,122 @@ void swift::rewriting::applyInverses(
     ASTContext &ctx,
     ArrayRef<Type> gps,
     ArrayRef<InverseRequirement> inverseList,
+    ArrayRef<StructuralRequirement> explicitRequirements,
```

Comment:
You can probably move this function into its own file because it’s horrible but also self-contained. That’s what I did with ConcreteContraction.cpp